packages/debug/src/node/debug-adapter-session.ts

```diff
@@ -64,18 +64,18 @@ export class DebugAdapterSessionImpl implements DebugAdapterSession {
         this.channel.onClose(() => this.channel = undefined);
 
         this.communicationProvider.output.on('data', (data: Buffer) => this.handleData(data));
-        this.communicationProvider.output.on('close', () => this.fireExited());
+        this.communicationProvider.output.on('close', () => this.onDebugAdapterExit(1, undefined)); // FIXME pass a proper exit code
```

Comment:
always `1` is not a real `exitCode`, a proper should come from the underlying process